CONTRIBUTING.md

```diff
@@ -335,6 +335,66 @@ since the latest release over the hotfixed `master`.
 This would mean, however, that the commits referenced from PRs previously merged to `develop` will no longer be part of `develop`'s history,
 which is rather unacceptable.
 
+### Plugin signing
+
+The valid non-expired key pair together with the password required for the plugin signing should be present in the CI environment.
+If they are absent, please take a look at the [plugin signing in IntelliJ](https://plugins.jetbrains.com/docs/intellij/plugin-signing.html#signing-methods) for updated instructions on how to do it.
+Currently, you would need to first generate the private key with:
+```shell
+export PLUGIN_SIGN_PRIVATE_KEY_PASS="Change2UrStr0ngP4ssword4PrivateKi"
+```
+```shell
+openssl genpkey\
+  -aes-256-cbc\
+  -algorithm RSA\
+  -out private.pem\
+  -pkeyopt rsa_keygen_bits:4096\
+  -pass env:PLUGIN_SIGN_PRIVATE_KEY_PASS
+```
+Then, use that for generating the certificate chain by running:
+```shell
+openssl req\
+  -key private.pem\
+  -new\
+  -x509\
+  -days 365\
+  -out chain.crt\
+  -passin env:PLUGIN_SIGN_PRIVATE_KEY_PASS\
+  -subj "/C=PL/ST=Krakow/L=Krakow/O=VirtusLab/OU=Git Machete team/CN=www.virtuslab.com/emailAddress=gitmachete@virtuslab.com"
+```
+
+#### PLugin signing as part of the CI publish process through the Gradle Plugin
+Please note that you would have to copy the contents of `private.pem`, `chain.crt`, and `$PASSWORD_ENVIRONMENT_VARIABLE_NAME`, in the [corresponding environment variables](https://app.circleci.com/settings/project/github/VirtusLab/git-machete-intellij-plugin); in order for the Gradle IntelliJ Plugin to pick them up and use them for the plugin signing task, before publishing to the Marketplace.
+For doing so, on a MacOS system you can follow the below instructions:
+
+1. Type the following command for copying the contents of the private key to the clipboard
+```shell
+pbcopy < private.pem
+```
+2. Create an environment variable named `PLUGIN_SIGN_PRIVATE_KEY` on the CI, and paste the content from the clipboard as its value.
+3. for copying the contents of the certificate to clipboard:
+```shell
+pbcopy < chain.crt
+```
+4. Create an environment variable named `PLUGIN_SIGN_CERT_CHAIN` on the CI, and paste the content from the clipboard as its value.
+5. Type the following command for copying the value of the private key password to the clipboard
+```shell
+echo $PLUGIN_SIGN_PRIVATE_KEY_PASS | pbcopy
+```
+6. Create an environment variable named `PLUGIN_SIGN_PRIVATE_KEY_PASS` on the CI and paste the contents of the clipboard as its value.
+
+#### local plugin signing for test
+
+You need to download the [IntelliJ signer cli](https://github.com/JetBrains/marketplace-zip-signer/releases). Then, following the [instructions](https://plugins.jetbrains.com/docs/intellij/plugin-signing.html#cli-tool) you should run:
+
+```shell
+java -jar marketplace-zip-signer-cli.jar sign\
+  -in "git-machete-intellij-plugin-version-SNAPSHOT+git.number.zip"\
+  -out "signed-machete-plugin.zip"\
```

Comment:
```suggestion
  -out "build/distributions/signed-machete-plugin.zip"\
```
To avoid spamming the main directory (as `*.zip` files aren't even git-ignored)

Reply:
(nvm, `./gradlew signPlugin` FTW)